core/domain/suggestion_services_test.py

```
@@ -865,12 +865,12 @@ def test_update_translation_suggestion_to_change_translation_html(
         self
     ) -> None:
         exploration = (
-            self.save_new_linear_exp_with_state_names_and_interactions(  # type: ignore[no-untyped-call]
+            self.save_new_linear_exp_with_state_names_and_interactions(
                 'exploration1', self.author_id, ['state 1'], ['TextInput'],
                 category='Algebra'))
         old_content = state_domain.SubtitledHtml(
             'content', '<p>old content html</p>').to_dict()
-        exploration.states['state 1'].update_content(
+        exploration.states['state 1'].update_content(  # type: ignore[no-untyped-call]
```

Why does this need to be added here?

---

Here `exploration.states` is a Dict whose keys are `state_names` and values are `state_domain.State` objects.

The `update_content` is one of the methods of `state_domain.State` object, but since state_domain is not fully type annotated yet  and `update_content` is defined in state_domain so `update_content` is also not type annotated yet.

So, until we typed the whole state_domain.py ( `update_content` method also -- because it is a part of state_domain ) we added this ignore statement -- to suppress the error because the suggestion_services_test.py file is annotated and any untyped call in the typed file throws an error.

we are not adding any comment here, because this ignore is going to be removed soon.

---

I hope I answered your question.

---

Gotcha. Makes sense. Thanks for the detailed explanation!